src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/ElasticSearchAnalysisGraph.java

```diff
@@ -203,6 +212,58 @@ public void construct() {
     nodeConfigClusterCollector.addAllUpstreams(Collections.singletonList(nodeConfigCollector));
     nodeConfigClusterCollector.addTag(TAG_AGGREGATE_UPSTREAM, LOCUS_DATA_NODE);
 
+    // Field Data Cache RCA
+    Metric fieldDataCacheEvictions = new Cache_FieldData_Eviction(EVALUATION_INTERVAL_SECONDS);
+    fieldDataCacheEvictions.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
+    addLeaf(fieldDataCacheEvictions);
+
+    Metric fieldDataCacheSizeGroupByOperation = new AggregateMetric(EVALUATION_INTERVAL_SECONDS,
+            Cache_FieldData_Size.NAME,
+            AggregateFunction.SUM,
+            MetricsDB.MAX, ShardStatsDerivedDimension.INDEX_NAME.toString());
+    fieldDataCacheSizeGroupByOperation.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
+    addLeaf(fieldDataCacheSizeGroupByOperation);
+
+    FieldDataCacheRca fieldDataCacheNodeRca = new FieldDataCacheRca(RCA_PERIOD,
+            fieldDataCacheEvictions,
+            fieldDataCacheSizeGroupByOperation);
+    fieldDataCacheNodeRca.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
+    fieldDataCacheNodeRca.addAllUpstreams(Collections.singletonList(fieldDataCacheEvictions));
+
+    FieldDataCacheClusterRca fieldDataCacheClusterRca = new FieldDataCacheClusterRca(RCA_PERIOD, fieldDataCacheNodeRca);
+    fieldDataCacheClusterRca.addTag(TAG_LOCUS, LOCUS_MASTER_NODE);
+    fieldDataCacheClusterRca.addAllUpstreams(Collections.singletonList(fieldDataCacheNodeRca));
+    fieldDataCacheClusterRca.addTag(TAG_AGGREGATE_UPSTREAM, LOCUS_DATA_NODE);
+
+    // Shard Request Cache RCA
+    Metric shardRequestCacheEvictions = new Cache_Request_Eviction(EVALUATION_INTERVAL_SECONDS);
+    shardRequestCacheEvictions.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
+    addLeaf(shardRequestCacheEvictions);
+    Metric shardRequestHits = new Cache_Request_Hit(EVALUATION_INTERVAL_SECONDS);
+    shardRequestHits.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
+    addLeaf(shardRequestHits);
+
+    Metric shardRequestCacheSizeGroupByOperation = new AggregateMetric(EVALUATION_INTERVAL_SECONDS,
+            Cache_Request_Size.NAME,
+            AggregateFunction.SUM,
+            MetricsDB.MAX, ShardStatsDerivedDimension.INDEX_NAME.toString());
+    shardRequestCacheSizeGroupByOperation.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
+    addLeaf(shardRequestCacheSizeGroupByOperation);
+
+    ShardRequestCacheRca shardRequestCacheNodeRca = new ShardRequestCacheRca(RCA_PERIOD,
+            shardRequestCacheEvictions,
+            shardRequestHits,
+            shardRequestCacheSizeGroupByOperation);
+    shardRequestCacheNodeRca.addTag(TAG_LOCUS, LOCUS_DATA_MASTER_NODE);
+    shardRequestCacheNodeRca.addAllUpstreams(Collections.singletonList(shardRequestHits));
```

Comment:
shardRequestCacheNodeRca needs to be connected to shardRequestCacheEvictions leaf node as well right?
Arrays.asList(All the metrics this node is consuming)?

Reply:
Thank You for catching this issue, we need to add remainder metrics as well.  Adding them.